aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,751 @@
+            # Expand dimension info
+            dim, size = dim_info
+
+            # If the dim was provided in the read dims we know a single plane for that
+            # dimension was requested so remove it
+            if dim in read_dims:
+                ops.append(0)
+            # Otherwise just read the full slice
+            else:
+                ops.append(slice(None, None, None))
+                real_dims.append(dim_info)
+
+        # Convert ops and run getitem
+        return data[tuple(ops)], real_dims
+
+    @staticmethod
+    def _imread(img: Path, offsets: List[np.ndarray], r_length: np.ndarray, read_dims: Optional[Dict[str, str]] = None) -> np.ndarray:
+        """
+        This function is a pass through to _read_image above
+        the difference is it returns the data without the dims structure.
+        Parameters
+        ----------
+        img: Path
+            Path to the LIF file to read.
+        offsets: List[numpy.ndarray]
+            A List of numpy ndarrays offsets, see _compute_offsets for more details.
+        r_length: numpy.ndarray
+            A 1D numpy array of read lengths, the index is the scene index
+        read_dims: Optional[Dict[str, int]]
+            The dimensions to read from the file as a dictionary of string to integer.
+            Default: None (Read all data from the image)
+
+        Returns
+        -------
+        data: np.ndarray
+            The data read for the dimensions provided.
+        """
+        data, dims = LifReader._read_image(img=img,
+                                           offsets=offsets,
+                                           r_length=r_length,
+                                           read_dims=read_dims
+                                           )
+        return data
+
+    @staticmethod
+    def _dims_shape(lif: LifFile):
+        """
+        Get the dimensions for the opened file from the binary data (not the metadata)
+
+        Parameters
+        ----------
+        lif: LifFile
+
+        Returns
+        -------
+        list[dict]
+            A list of dictionaries containing Dimension / depth. If the shape is consistent across Scenes then
+            the list will have only one Dictionary. If the shape is inconsistent the the list will have a dictionary
+             for each Scene. A consistently shaped file with 3 scenes, 7 time-points
+            and 4 Z slices containing images of (h,w) = (325, 475) would return
+            [
+             {'S': (0, 3), 'T': (0,7), 'X': (0, 475), 'Y': (0, 325), 'Z': (0, 4)}
+            ].
+            The result for a similarly shaped file but with different number of time-points per scene would yield
+            [
+             {'S': (0, 1), 'T': (0,8), 'X': (0, 475), 'Y': (0, 325), 'Z': (0, 4)},
+             {'S': (1, 2), 'T': (0,6), 'X': (0, 475), 'Y': (0, 325), 'Z': (0, 4)},
+             {'S': (2, 3), 'T': (0,7), 'X': (0, 475), 'Y': (0, 325), 'Z': (0, 4)}
+            ]
+
```

Comment:
I am like 99% sure this will break on documentation render. Can you try running `make serve-docs` and see what errors you get. Fix formatting until no errors arise.